### PR TITLE
EVG-12614 script to sanitize json

### PR DESCRIPTION
**Pull request by john-m-liu**
run it like `make scramble file=cmd/scrambled-eggs/test.json`

This isn't written in go because I need loose type rules. This isn't written in js/ts because I don't want to introduce a nodejs dependency

### CODE REVIEW

File: cmd/scrambled-eggs/scramble.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+                    value[i] = traverse(v)
+                obj[key] = value
+            else:
+                if (isinstance(value, str) or isinstance(value, unicode)):
+                    obj[key] = hashlib.sha256(value).hexdigest()
+    else:
+        if (isinstance(obj, str) or isinstance(obj, unicode)):
+            return hashlib.sha256(obj).hexdigest()
+
+
```
Comment (bsamek) [inline cmd/scrambled-eggs/scramble.py:37]:
```
if __name__ == "__main__":
    main()
```

File: cmd/scrambled-eggs/scramble.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import json
+import argparse
+import hashlib
+
+# this script takes a json file as input and destructively hashes all string values in it
+def main():
+    args = argparse.ArgumentParser()
```
Comment (bsamek) [inline cmd/scrambled-eggs/scramble.py:7]: Put the docstring here is triple quotes.

File: cmd/scrambled-eggs/scramble.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    args.add_argument("file")
+    flags = args.parse_args()
+
+    with open(flags.file, "r") as f:
+        data = json.load(f)
+        f.close()
```
Comment (bsamek) [inline cmd/scrambled-eggs/scramble.py:13]: This is redundant, since you're using `with`.

File: cmd/scrambled-eggs/scramble.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+    traverse(data)
+    
+    with open(flags.file, "w") as f:
+        json.dump(data, f)
+        f.close()
```
Comment (bsamek) [inline cmd/scrambled-eggs/scramble.py:19]: redundant with `with`

File: cmd/scrambled-eggs/scramble.py [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+        for key, value in obj.items():
+            if isinstance(value, dict):
+                traverse(value)
+            elif isinstance(value, list) or isinstance(value, tuple):
+                for i,v in enumerate(value):
+                    value[i] = traverse(v)
+                obj[key] = value
+            else:
+                if (isinstance(value, str) or isinstance(value, unicode)):
+                    obj[key] = hashlib.sha256(value).hexdigest()
```
Comment (hhoke) [inline cmd/scrambled-eggs/scramble.py:30]: I'm assuming the keys never need to be sanitized?

Comment (john-m-liu) [reply cmd/scrambled-eggs/scramble.py:30]: It's more that changing keys will make the data unusable. In general we don't store data in keys, though there are exceptions

**Comment by tgrander** [comment]
Screen Shot 2020-07-22 at 1 37 47 PM

i get this error when running it on my json file

**Comment by hhoke** [comment]
Was able to recreate, working on a fix

**Comment by hhoke** [comment]
https://github.com/evergreen-ci/evergreen/commit/d710463aab6cc91e7b896a9e5ca0c454bee5b4cd

**Pull request merged**